src/ServiceMap.php

```
@@ -276,13 +328,12 @@
                 }
 
                 if (is_object($definition)) {
-                    $this->components[$id] = get_class($definition);
-
+                    $this->components[$id] = $definition;
                     continue;
                 }
 
                 if (isset($definition['class']) && is_string($definition['class']) && $definition['class'] !== '') {
-                    $this->components[$id] = $definition['class'];
+                    $this->components[$id] = $definition;
```

PHPStan (2.1.15)

Property yii2\extensions\phpstan\ServiceMap::$components (array<string, array<string, mixed>|object>) does not accept array<string, array|object>.


(assign.propertyType)